File: How To Find Person/Ruby/WhitePages-PersonLookup/app/models/result.rb
class Result

  attr_accessor :json_response, :data_parse, :error, :person

  def initialize(json_response)
    @json_response = json_response
    @data_parse = data_parse
  end

  def data_parse
    unless json_response['error'].nil?
      @error = json_response['error']['message']
    else
      @person = json_response['results'].map do |id|
        person = Person.new(json_response['dictionary'][id])
        best_location = person.best_location
        person.location = Location.new(json_response['dictionary'][best_location]).data if best_location
        person.data
      end.reject(&:blank?)
    end
  end
end

File: How To Find Person/PHP/WhitePages-PersonLookup/lib/result.php
<?php
/**
 * Library for abstraction of methods to use Whitepages Pro API
 * @require     PHP5
 *
 * @author      Kushal Shah
 * @date        2014-06-02
 */

include 'person.php';

class Result
{
    public $json_response;
    public $error;
    public $person;


    public function __construct($json_response)
    {
        $this->json_response = $json_response;
        $this->data_parse();
    }

    public function data_parse()
    {
        if (!empty($this->json_response['error'])) {
            $this->error = $this->json_response['error']['message'];
        } else {
            $person_array = array();
            while (list(, $results_val) = each($this->json_response['results'])) {
                if (!empty($results_val)) {
                    $person = new Person($this->json_response['dictionary'][$results_val]);
                    $best_location = $person->best_location();
                    if (!empty($best_location)) {
                        $person->location($this->json_response['dictionary'][$best_location]);
                    }
                    array_push($person_array, $person->data());
                }
            }
            $this->person = $person_array;
        }
    }

}

